### PR TITLE
#167271157 setup card divider molecule

### PR DESCRIPTION
**Pull request by frankly034**
#### What does this PR do?

- This PR sets up a CardDividier molecule with styleguide

#### Description of task to be completed?

- setup carddivider molecule
- setup styleguidist for the carddivider
- write tests for the carddivider molecule
- add title font size variable attribute
- refactor divider to display inline
- add webpack molecule module resolver
- add jest config molecule module resolver
- add molecule sections to styleguide configuration

#### How should this be manually tested?

  - $ git clone https://github.com/andela/king-kong-ah-frontend.git
  - $ npm install
  - $ npm test

#### Any background context you want to provide?

- You would need npm to run this application

#### What are the relevant pivotal tracker stories?

- PT Story link - [#167271157](https://www.pivotaltracker.com/story/show/167271157)

#### Screenshots (if appropriate)

image

### CODE REVIEW

**Review by akinyeleolat — APPROVED** [review]

**Review by Sojisoyoye — APPROVED** [review]

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ * @prop {string} textPadding - text padding
+ * @prop {string} content - content
+ * @prop {string} cardDividerPadding - card divider padding
+ * @prop {string} display - container display
+ *
+ * @return {component} CardDivider
+ */
+const CardDivider = ({
+  text,
+  cardDividerMargin,
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: Why is this needed, we already have a margin

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  width,
+  height,
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: The height prop is not needed as there is no where else in the design that specifies this should be a different height

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  margin,
+  dividerColor,
+  fontSize,
+  textAlign,
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: Is there a place in the design that the text is in anywhere other than the center

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  textColor,
+  textDisplay,
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: Why is this display different from the normal display

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  textTransform,
+  fontWeight,
+  textPadding,
+  content,
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: What is this content doing ?

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  display,
+  cardDividerPadding,
+}) => {
+  const SingleDivider = () => (
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: I expect these to be named like function e.g. `renderDivider()`

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    <Divider
+      width={width}
+      height={height}
+      margin={margin}
+      dividerColor={dividerColor}
+    />
+  );
+  const DividerText = () => (
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: Both render functions should be below the return

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  );
+
+  return (
+    <FlexContainer
+      display={display}
+      margin={cardDividerMargin}
+      padding={cardDividerPadding}
+    >
+      <SingleDivider />
+      {text && (
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: What is the difference between text and DividerText

File: src/components/UI/molecules/CardDivider/CardDivider.spec.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import { render, cleanup } from '<src>/helpers/testUtils';
+import CardDivider from './CardDivider';
+import '@testing-library/jest-dom/extend-expect';
+
+afterEach(cleanup);
+
+describe('CardDivider', () => {
+  it('should render the CardDivider component', () => {
+    const { container } = render(<CardDivider />);
+
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.spec.js:11]: This is not enough tests considering the number of props you are using

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: src/components/UI/molecules/CardDivider/CardDivider.jsx (hunk 8 above)
Comment (frankly034) [reply src/components/UI/molecules/CardDivider/CardDivider.jsx]: `text` is a prop that is passed into the CardDivider. It is rendered in the Text component of the CardDivider. `DividerText` is actually a function that renders the Text component. I've renamed it to renderText().

File: src/components/UI/molecules/CardDivider/CardDivider.jsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  fontSize,
+  display,
+}) => {
+  return (
+    <CardDivider.Container>
+      <FlexContainer
+        display={display}
+        margin={margin}
+      >
+        { renderDivider() }
```
Comment (benfluleck) [inline src/components/UI/molecules/CardDivider/CardDivider.jsx]: I don't think there should be space around the braces, please configure this rule in eslint

**Review by benfluleck — APPROVED** [review]
